### PR TITLE
Hardening day one

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedPropertiesWsFilter.java [hunk 1]
```diff
@@ -245,6 +248,9 @@ private void addParameterIfPresent(String parameterKey, Optional<String> value)
       value.ifPresent(s -> additionalParams.put(parameterKey, s));
     }
 
+    private void failIfIdIsNull(Optional<String> id) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedPropertiesWsFilter.java]: MAJOR Make "failIfIdIsNull" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)
MINOR Specify a "String" parameter instead. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553)

**Comment by SonarTech** [comment]
SonarQube analysis reported 13 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 12 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [DeprecatedPropertiesWsFilter.java#L162](https://github.com/SonarSource/sonarqube/blob/bb5b371e0caec635c8b89a768b16d2363aa1bd48/server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedPropertiesWsFilter.java#L162): Specify a "String" parameter instead. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553)
1. ![MINOR][MINOR] [DeprecatedPropertiesWsFilter.java#L162](https://github.com/SonarSource/sonarqube/blob/bb5b371e0caec635c8b89a768b16d2363aa1bd48/server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedPropertiesWsFilter.java#L162): Specify a "String" parameter instead. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553)
1. ![MINOR][MINOR] [DeprecatedPropertiesWsFilter.java#L170](https://github.com/SonarSource/sonarqube/blob/bb5b371e0caec635c8b89a768b16d2363aa1bd48/server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedPropertiesWsFilter.java#L170): Specify a "String" parameter instead. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553)
1. ![MINOR][MINOR] [DeprecatedPropertiesWsFilter.java#L170](https://github.com/SonarSource/sonarqube/blob/bb5b371e0caec635c8b89a768b16d2363aa1bd48/server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedPropertiesWsFilter.java#L170): Specify a "String" parameter instead. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553)
1. ![MINOR][MINOR] [DeprecatedPropertiesWsFilter.java#L174](https://github.com/SonarSource/sonarqube/blob/bb5b371e0caec635c8b89a768b16d2363aa1bd48/server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedPropertiesWsFilter.java#L174): Specify a "String" parameter instead. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553)
1. ![MINOR][MINOR] [DeprecatedPropertiesWsFilter.java#L174](https://github.com/SonarSource/sonarqube/blob/bb5b371e0caec635c8b89a768b16d2363aa1bd48/server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedPropertiesWsFilter.java#L174): Specify a "String" parameter instead. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553)
1. ![MINOR][MINOR] [DeprecatedPropertiesWsFilter.java#L228](https://github.com/SonarSource/sonarqube/blob/bb5b371e0caec635c8b89a768b16d2363aa1bd48/server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedPropertiesWsFilter.java#L228): Specify a "String" parameter instead. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553)
1. ![MINOR][MINOR] [DeprecatedPropertiesWsFilter.java#L228](https://github.com/SonarSource/sonarqube/blob/bb5b371e0caec635c8b89a768b16d2363aa1bd48/server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedPropertiesWsFilter.java#L228): Specify a "String" parameter instead. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553)
1. ![MINOR][MINOR] [DeprecatedPropertiesWsFilter.java#L240](https://github.com/SonarSource/sonarqube/blob/bb5b371e0caec635c8b89a768b16d2363aa1bd48/server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedPropertiesWsFilter.java#L240): Specify a "String" parameter instead. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553)
1. ![MINOR][MINOR] [DeprecatedPropertiesWsFilter.java#L240](https://github.com/SonarSource/sonarqube/blob/bb5b371e0caec635c8b89a768b16d2363aa1bd48/server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedPropertiesWsFilter.java#L240): Specify a "String" parameter instead. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: server/sonar-server/src/test/java/org/sonar/server/test/ws/ListActionTest.java [hunk 2]
```diff
@@ -31,280 +31,255 @@
...
   @Rule
   public ExpectedException expectedException = ExpectedException.none();
   @Rule
+  public EsTester es = new EsTester(new TestIndexDefinition(new MapSettings()));
+  @Rule
   public UserSessionRule userSessionRule = UserSessionRule.standalone();
   @Rule
-  public DbTester db = DbTester.create(System2.INSTANCE)
-      .setDefaultOrganizationUuid(DEFAULT_ORGANIZATION_UUID);
+  public DbTester db = DbTester.create(System2.INSTANCE);
```
Comment (teryk) [inline server/sonar-server/src/test/java/org/sonar/server/test/ws/ListActionTest.java]: could you do the same thing for the constructor `DbTester.create()` if it's not already done?

File: server/sonar-server/src/test/java/org/sonar/server/test/ws/ListActionTest.java [hunk 3]
```diff
@@ -31,280 +31,255 @@
...
-    dbClient.componentDao().insert(db.getSession(), TestFile1.dto());
-    db.getSession().commit();
+  public void return_test() throws Exception {
+    userSessionRule.addProjectUuidPermissions(CODEVIEWER, project.uuid());
+    DbFileSources.Test test = newTest(mainFile, 10, 11, 12, 20, 21, 25).setStatus(OK).build();
+    insertTests(testFile, test);
+
+    ListResponse request = call(ws.newRequest().setParam(TEST_ID, test.getUuid()));
+
+    assertThat(request.getTestsList()).hasSize(1);
```
Comment (teryk) [inline server/sonar-server/src/test/java/org/sonar/server/test/ws/ListActionTest.java]: comment: I think we should make a better use of custom assertJ assertion, at least we should try

File: server/sonar-server/src/test/java/org/sonar/server/test/ws/ListActionTest.java [hunk 4]
```diff
@@ -31,280 +31,255 @@
...
+    insertTests(anotherTestFile, test3);
+
+    ListResponse request = call(ws.newRequest().setParam(TEST_FILE_ID, testFile.uuid()));
+
+    assertThat(request.getTestsList()).extracting(WsTests.Test::getId).containsOnly(test1.getUuid(), test2.getUuid());
+  }
 
-    WsTester.TestRequest request = ws.newGetRequest("api/tests", "list").setParam(ListAction.TEST_FILE_ID, TestFile1.FILE_UUID);
+  @Test
+  public void return_tests_based_on_test_file_key() throws Exception {
```
Comment (teryk) [inline server/sonar-server/src/test/java/org/sonar/server/test/ws/ListActionTest.java]: why prefix with `return` all the time? for me it's the same as the `test` prefix

File: server/sonar-server/src/test/java/org/sonar/server/test/ws/ListActionTest.java [hunk 5]
```diff
@@ -31,280 +31,255 @@
...
-  public void fail_when_main_file_uuid_without_line_number() throws Exception {
-    ws.newGetRequest("api/tests", "list").setParam(ListAction.SOURCE_FILE_ID, "ANY-UUID").execute();
-  }
+  @Test
+  public void fail_when_not_enough_privilege_on_test_uuid() throws Exception {
+    userSessionRule.addProjectUuidPermissions(USER, project.uuid());
+    DbFileSources.Test test = newTest(mainFile, 10).build();
+    insertTests(testFile, test);
 
-  @Test(expected = ForbiddenException.class)
```
Comment (teryk) [inline server/sonar-server/src/test/java/org/sonar/server/test/ws/ListActionTest.java]: 👍

File: server/sonar-server/src/test/java/org/sonar/server/test/ws/ListActionTest.java (hunk 2 above)
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/test/ws/ListActionTest.java]: What do you mean ? Replacing DbTester.create(System2.INSTANCE) by DbTester.create() ?

File: server/sonar-server/src/test/java/org/sonar/server/test/ws/ListActionTest.java (hunk 3 above)
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/test/ws/ListActionTest.java]: WDYM ? What is missing ?

File: server/sonar-server/src/test/java/org/sonar/server/test/ws/ListActionTest.java (hunk 4 above)
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/test/ws/ListActionTest.java]: I don't know, I'm trying to find the best sentence that is describing what is tested, but it's not easy for me

File: server/sonar-server/src/main/java/org/sonar/server/exceptions/Errors.java [hunk 6]
```diff
@@ -70,20 +70,12 @@ public boolean check(boolean expression, String l10nKey, Object... l10nParams) {
...
-  public void writeJsonAsWarnings(JsonWriter json, I18n i18n, Locale locale) {
-    writeJson(json, i18n, locale, "warnings");
-  }
-
-  private void writeJson(JsonWriter json, I18n i18n, Locale locale, String name) {
+  public void writeJson(JsonWriter json, I18n i18n) {
     if (!messages.isEmpty()) {
-      json.name(name).beginArray();
+      json.name("errors").beginArray();
       for (Message message : messages) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/exceptions/Errors.java]: MINOR Remove this use of "Message"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedPropertiesWsFilter.java [hunk 7]
```diff
@@ -245,6 +248,9 @@ private void addParameterIfPresent(String parameterKey, Optional<String> value)
       value.ifPresent(s -> additionalParams.put(parameterKey, s));
     }
 
+    private static void failIfIdIsNull(Optional<String> id) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedPropertiesWsFilter.java]: MINOR Specify a "String" parameter instead. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553)

File: server/sonar-server/src/main/java/org/sonar/server/exceptions/BadRequestException.java [hunk 8]
```diff
@@ -32,11 +32,19 @@
...
   public BadRequestException(String l10nKey, Object... l10nParams) {
     super(HTTP_BAD_REQUEST);
     this.errors = new Errors().add(Message.of(l10nKey, l10nParams));
   }
 
+  /**
+   * @deprecated Bundle key should be replaced by real message as only english is returned on server side errors
+   */
+  @Deprecated
   public BadRequestException(List<Message> messages) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/exceptions/BadRequestException.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: server/sonar-server/src/main/java/org/sonar/server/exceptions/BadRequestException.java [hunk 9]
```diff
@@ -32,11 +32,19 @@
 
   private final transient Errors errors;
 
+  /**
+   * @deprecated Bundle key should be replaced by real message as only english is returned on server side errors
+   */
+  @Deprecated
   public BadRequestException(String l10nKey, Object... l10nParams) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/exceptions/BadRequestException.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: server/sonar-server/src/main/java/org/sonar/server/exceptions/Message.java [hunk 10]
```diff
@@ -20,13 +20,15 @@
...
 import javax.annotation.Nullable;
+import org.apache.commons.lang.StringUtils;
 
-import java.util.Arrays;
-
+/**
+ * @deprecated Bundle key should be replaced by real message as only english is returned on server side messages
+ */
+@Deprecated
 public class Message {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/exceptions/Message.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 31 issues
* ![MAJOR][MAJOR] 4 major
* ![MINOR][MINOR] 24 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ResourcesWs.java#L36](https://github.com/SonarSource/sonarqube/blob/cd7449916974667c2a9b838c21091b11427e5b33/server/sonar-server/src/main/java/org/sonar/server/component/ws/ResourcesWs.java#L36): Make "defineIndexAction" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)
1. ![MAJOR][MAJOR] [TimeMachineWs.java#L36](https://github.com/SonarSource/sonarqube/blob/cd7449916974667c2a9b838c21091b11427e5b33/server/sonar-server/src/main/java/org/sonar/server/measure/ws/TimeMachineWs.java#L36): Make "defineIndexAction" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)
1. ![MAJOR][MAJOR] [UserPropertiesWs.java#L36](https://github.com/SonarSource/sonarqube/blob/cd7449916974667c2a9b838c21091b11427e5b33/server/sonar-server/src/main/java/org/sonar/server/user/ws/UserPropertiesWs.java#L36): Make "defineIndexAction" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)
1. ![MAJOR][MAJOR] [WebServiceEngine.java#L65](https://github.com/SonarSource/sonarqube/blob/cd7449916974667c2a9b838c21091b11427e5b33/server/sonar-server/src/main/java/org/sonar/server/ws/WebServiceEngine.java#L65): Remove this unused "userSession" private field. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1068)
1. ![MINOR][MINOR] [BadRequestException.java#L62](https://github.com/SonarSource/sonarqube/blob/cd7449916974667c2a9b838c21091b11427e5b33/server/sonar-server/src/main/java/org/sonar/server/exceptions/BadRequestException.java#L62): Remove this use of "Message"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [BadRequestException.java#L70](https://github.com/SonarSource/sonarqube/blob/cd7449916974667c2a9b838c21091b11427e5b33/server/sonar-server/src/main/java/org/sonar/server/exceptions/BadRequestException.java#L70): Remove this use of "Message"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [Errors.java#L35](https://github.com/SonarSource/sonarqube/blob/cd7449916974667c2a9b838c21091b11427e5b33/server/sonar-server/src/main/java/org/sonar/server/exceptions/Errors.java#L35): Remove this use of "Message"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [Errors.java#L45](https://github.com/SonarSource/sonarqube/blob/cd7449916974667c2a9b838c21091b11427e5b33/server/sonar-server/src/main/java/org/sonar/server/exceptions/Errors.java#L45): Remove this use of "Message"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [Errors.java#L46](https://github.com/SonarSource/sonarqube/blob/cd7449916974667c2a9b838c21091b11427e5b33/server/sonar-server/src/main/java/org/sonar/server/exceptions/Errors.java#L46): Remove this use of "Message"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [Errors.java#L52](https://github.com/SonarSource/sonarqube/blob/cd7449916974667c2a9b838c21091b11427e5b33/server/sonar-server/src/main/java/org/sonar/server/exceptions/Errors.java#L52): Remove this use of "Message"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 31 issues
* ![MAJOR][MAJOR] 4 major
* ![MINOR][MINOR] 24 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ResourcesWs.java#L36](https://github.com/SonarSource/sonarqube/blob/6e8d66e79154e11a9b54bd8e0e10df4c7c3c6032/server/sonar-server/src/main/java/org/sonar/server/component/ws/ResourcesWs.java#L36): Make "defineIndexAction" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)
1. ![MAJOR][MAJOR] [TimeMachineWs.java#L36](https://github.com/SonarSource/sonarqube/blob/6e8d66e79154e11a9b54bd8e0e10df4c7c3c6032/server/sonar-server/src/main/java/org/sonar/server/measure/ws/TimeMachineWs.java#L36): Make "defineIndexAction" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)
1. ![MAJOR][MAJOR] [UserPropertiesWs.java#L36](https://github.com/SonarSource/sonarqube/blob/6e8d66e79154e11a9b54bd8e0e10df4c7c3c6032/server/sonar-server/src/main/java/org/sonar/server/user/ws/UserPropertiesWs.java#L36): Make "defineIndexAction" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)
1. ![MAJOR][MAJOR] [WebServiceEngine.java#L65](https://github.com/SonarSource/sonarqube/blob/6e8d66e79154e11a9b54bd8e0e10df4c7c3c6032/server/sonar-server/src/main/java/org/sonar/server/ws/WebServiceEngine.java#L65): Remove this unused "userSession" private field. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1068)
1. ![MINOR][MINOR] [BadRequestException.java#L62](https://github.com/SonarSource/sonarqube/blob/6e8d66e79154e11a9b54bd8e0e10df4c7c3c6032/server/sonar-server/src/main/java/org/sonar/server/exceptions/BadRequestException.java#L62): Remove this use of "Message"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [BadRequestException.java#L70](https://github.com/SonarSource/sonarqube/blob/6e8d66e79154e11a9b54bd8e0e10df4c7c3c6032/server/sonar-server/src/main/java/org/sonar/server/exceptions/BadRequestException.java#L70): Remove this use of "Message"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [Errors.java#L35](https://github.com/SonarSource/sonarqube/blob/6e8d66e79154e11a9b54bd8e0e10df4c7c3c6032/server/sonar-server/src/main/java/org/sonar/server/exceptions/Errors.java#L35): Remove this use of "Message"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [Errors.java#L45](https://github.com/SonarSource/sonarqube/blob/6e8d66e79154e11a9b54bd8e0e10df4c7c3c6032/server/sonar-server/src/main/java/org/sonar/server/exceptions/Errors.java#L45): Remove this use of "Message"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [Errors.java#L46](https://github.com/SonarSource/sonarqube/blob/6e8d66e79154e11a9b54bd8e0e10df4c7c3c6032/server/sonar-server/src/main/java/org/sonar/server/exceptions/Errors.java#L46): Remove this use of "Message"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [Errors.java#L52](https://github.com/SonarSource/sonarqube/blob/6e8d66e79154e11a9b54bd8e0e10df4c7c3c6032/server/sonar-server/src/main/java/org/sonar/server/exceptions/Errors.java#L52): Remove this use of "Message"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 31 issues
* ![MAJOR][MAJOR] 4 major
* ![MINOR][MINOR] 24 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ResourcesWs.java#L36](https://github.com/SonarSource/sonarqube/blob/37868d0d6f9c3360695f4ed908854cb2cc0235cb/server/sonar-server/src/main/java/org/sonar/server/component/ws/ResourcesWs.java#L36): Make "defineIndexAction" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)
1. ![MAJOR][MAJOR] [TimeMachineWs.java#L36](https://github.com/SonarSource/sonarqube/blob/37868d0d6f9c3360695f4ed908854cb2cc0235cb/server/sonar-server/src/main/java/org/sonar/server/measure/ws/TimeMachineWs.java#L36): Make "defineIndexAction" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)
1. ![MAJOR][MAJOR] [UserPropertiesWs.java#L36](https://github.com/SonarSource/sonarqube/blob/37868d0d6f9c3360695f4ed908854cb2cc0235cb/server/sonar-server/src/main/java/org/sonar/server/user/ws/UserPropertiesWs.java#L36): Make "defineIndexAction" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)
1. ![MAJOR][MAJOR] [WebServiceEngine.java#L65](https://github.com/SonarSource/sonarqube/blob/37868d0d6f9c3360695f4ed908854cb2cc0235cb/server/sonar-server/src/main/java/org/sonar/server/ws/WebServiceEngine.java#L65): Remove this unused "userSession" private field. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1068)
1. ![MINOR][MINOR] [BadRequestException.java#L62](https://github.com/SonarSource/sonarqube/blob/37868d0d6f9c3360695f4ed908854cb2cc0235cb/server/sonar-server/src/main/java/org/sonar/server/exceptions/BadRequestException.java#L62): Remove this use of "Message"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [BadRequestException.java#L70](https://github.com/SonarSource/sonarqube/blob/37868d0d6f9c3360695f4ed908854cb2cc0235cb/server/sonar-server/src/main/java/org/sonar/server/exceptions/BadRequestException.java#L70): Remove this use of "Message"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [Errors.java#L35](https://github.com/SonarSource/sonarqube/blob/37868d0d6f9c3360695f4ed908854cb2cc0235cb/server/sonar-server/src/main/java/org/sonar/server/exceptions/Errors.java#L35): Remove this use of "Message"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [Errors.java#L45](https://github.com/SonarSource/sonarqube/blob/37868d0d6f9c3360695f4ed908854cb2cc0235cb/server/sonar-server/src/main/java/org/sonar/server/exceptions/Errors.java#L45): Remove this use of "Message"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [Errors.java#L46](https://github.com/SonarSource/sonarqube/blob/37868d0d6f9c3360695f4ed908854cb2cc0235cb/server/sonar-server/src/main/java/org/sonar/server/exceptions/Errors.java#L46): Remove this use of "Message"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [Errors.java#L52](https://github.com/SonarSource/sonarqube/blob/37868d0d6f9c3360695f4ed908854cb2cc0235cb/server/sonar-server/src/main/java/org/sonar/server/exceptions/Errors.java#L52): Remove this use of "Message"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Pull request merged**